### PR TITLE
[MIG][10.0][Gattlen - t6396] Migrating jasper_server to 10.0

### PR DESCRIPTION
**Pull request by BT-aestebanez**

### CODE REVIEW

File: jasper_server/common/jasper.py [hunk 1]
```diff
@@ -43,17 +38,15 @@ class report_jasper(report_int):
     Extend report_int to use Jasper Server
     """
 
-    def create(self, cr, uid, ids, data, context=None):
-        if context is None:
-            context = {}
+    def create(self, cr, uid, res_ids, data, context=None):
```
Comment (BT-rmartin) [inline jasper_server/common/jasper.py:41]: This is old API. Is it needed for some reason?

File: jasper_server/common/__init__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# b-*- encoding: utf-8 -*-
```
Comment (BT-rmartin) [inline jasper_server/common/__init__.py:1]: Is this file needed? You are not importing anything and you deleted common.py

File: jasper_server/common/jasper.py (hunk 1 above)
Comment (BT-aestebanez) [reply jasper_server/common/jasper.py:41]: Yes, as you can see, the class inherits from report_int (it is not a model) and this report_int defines its create method as follows:
def create(self, cr, uid, ids, datas, context=None):
https://github.com/brain-tec/odoo/blob/10.0/odoo/report/interface.py#L47

File: jasper_server/common/jasperlib.py [hunk 3]
```diff
@@ -32,9 +32,18 @@
 import os
 
 KNOWN_FORMAT = [
-    'PDF', 'XLS', 'XLSX', 'HTML', 'RTF',
-    'CSV', 'XML', 'DOCX', 'ODT', 'ODS',
-    'JPRINT'
+    # TODO: Some formats are commented because currently they are not working (v10.0.1.0)
```
Comment (BT-rmartin) [inline jasper_server/common/jasperlib.py:35]: Please, open a pending task to remember this.

File: jasper_server/data/jasper_document_extension_data.xml [hunk 4]
```diff
@@ -24,65 +24,65 @@
...
         <record model="jasper.document.extension" id="jasper_document_extension_1">
-            <field name="jasper_code">ODS</field>
             <field name="name">OpenOffice Writer</field>
-            <field name="extension">ods</field>
+            <field name="extension">odt</field>
         </record>
+        -->
 
         <record model="jasper.document.extension" id="jasper_document_extension_2">
-            <field name="jasper_code">PDF</field>
```
Comment (BT-rmartin) [inline jasper_server/data/jasper_document_extension_data.xml:34]: Why are you deleting jasper codes?

File: jasper_server/common/__init__.py (hunk 2 above)
Comment (BT-aestebanez) [reply jasper_server/common/__init__.py:1]: Yes
Actually, I don't know why, but without this file, imports of files from this folder do not work:

Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/werkzeug/serving.py", line 180, in run_wsgi
    execute(self.server.app)
  File "/usr/lib/python2.7/dist-packages/werkzeug/serving.py", line 168, in execute
    application_iter = app(environ, start_response)
  File "/home/esal1/Gattlen_v10/git_repositories/odoo/odoo/service/server.py", line 246, in app
    return self.app(e, s)
  File "/home/esal1/Gattlen_v10/git_repositories/odoo/odoo/service/wsgi_server.py", line 184, in application
    return application_unproxied(environ, start_response)
  File "/home/esal1/Gattlen_v10/git_repositories/odoo/odoo/service/wsgi_server.py", line 170, in application_unproxied
    result = handler(environ, start_response)
  File "/home/esal1/Gattlen_v10/git_repositories/odoo/odoo/http.py", line 1305, in __call__
    self.load_addons()
  File "/home/esal1/Gattlen_v10/git_repositories/odoo/odoo/http.py", line 1326, in load_addons
    m = __import__('odoo.addons.' + module)
  File "/home/esal1/Gattlen_v10/git_repositories/odoo/odoo/modules/module.py", line 81, in load_module
    execfile(modfile, new_mod.__dict__)
  File "/home/esal1/Gattlen_v10/git_repositories/openerp-jasperserver/jasper_server/__init__.py", line 25, in <module>
    import models
  File "/home/esal1/Gattlen_v10/git_repositories/openerp-jasperserver/jasper_server/models/__init__.py", line 23, in <module>
    import jasper_server
  File "/home/esal1/Gattlen_v10/git_repositories/openerp-jasperserver/jasper_server/models/jasper_server.py", line 29, in <module>
    from odoo.addons.jasper_server.common import jasperlib
ImportError: No module named common

File: jasper_server/ir_actions.py [hunk 5]
```diff
@@ -1,113 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (BT-rmartin) [inline jasper_server/ir_actions.py:1]: Is this file only needed for Jasper reports based on RML and Webkit reports?

File: jasper_server/common/jasperlib.py (hunk 3 above)
Comment (BT-aestebanez) [reply jasper_server/common/jasperlib.py:35]: Ok,
I've created a new task on the brain-tec project: t6404

File: jasper_server/ir_actions.py (hunk 5 above)
Comment (BT-rmartin) [reply jasper_server/ir_actions.py:1]: This is related to new file jasper_server/models/ir_actions_report_xml_ext.py

File: jasper_server/data/jasper_document_extension_data.xml (hunk 4 above)
Comment (BT-aestebanez) [reply jasper_server/data/jasper_document_extension_data.xml:34]: It was redundant, if you take a look at the extension and jasper_code, they are similar but the jasper_code is on upper_case.
Moreover, this field was being used only to set the values of the Selection field from a jasper.document, but now I changed the selection for a many2one (which I think is more correct), and it was not used anymore.
I think is better to delete it to avoid duplicity and a source of errors

File: jasper_server/models/jasper_document_label.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+class jasper_document_label(models.Model):
+    _name = 'jasper.document.label'
+    _description = 'Manage label in document, for different language'
+
+    name = fields.Char('Parameter', size=64,
+                       help='Name of the parameter send to JasperServer, prefix with I18N_\n'
+                            'eg: test become I18N_TEST as parameter', required=True)
+    value = fields.Char('Value', size=256, required=True, translate=True,
+                        help='Name of the label, this field must be translate in all '
```
Comment (BT-rmartin) [inline jasper_server/models/jasper_document_label.py]: translated

File: jasper_server/models/jasper_document_label.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+from odoo import fields, models
+
+
+class jasper_document_label(models.Model):
+    _name = 'jasper.document.label'
+    _description = 'Manage label in document, for different language'
+
+    name = fields.Char('Parameter', size=64,
+                       help='Name of the parameter send to JasperServer, prefix with I18N_\n'
```
Comment (BT-rmartin) [inline jasper_server/models/jasper_document_label.py]: that is sent

File: jasper_server/wizard/load_file_view.xml [hunk 8]
```diff
@@ -1,19 +1,20 @@
...
                 <group colspan="2">
-                    <field name="datafile"/>
+                    <field name="datafile" widget="binary" filename="name"/>
+                    <field name="save_as_attachment" />
+                    <field name="name" invisible="True" />
                 </group>
                 <footer>
-                    <button type="special" special="cancel" string="Cancel" icon="gtk-cancel"/>
-                    <button type="object" string="Next" name="import_file" icon="gtk-ok"/>
+                    <button special="cancel" string="Cancel" icon="fa-times-circle text-danger"/>
```
Comment (BT-rmartin) [inline jasper_server/wizard/load_file_view.xml:16]: type="special" is not needed anymore?

**Comment by BT-rmartin** [comment]
I finished the review. Good job!

**Review by BT-rmartin — APPROVED** [review]
I finished the review. Good job!

File: jasper_server/wizard/load_file_view.xml (hunk 8 above)
Comment (BT-aestebanez) [reply jasper_server/wizard/load_file_view.xml:16]: Yes, now it seems it is not needed

File: jasper_server/ir_actions.py (hunk 5 above)
Comment (BT-aestebanez) [reply jasper_server/ir_actions.py:1]: It seems it's used for all the reports
https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/ir/ir_actions.py#L87

**Comment by BT-aestebanez** [comment]
Thanks for the review 
:)
